src/Castle.Windsor/MicroKernel/Releasers/LifecycledComponentsReleasePolicy.cs

```diff
@@ -185,19 +210,31 @@ public virtual void Track(object instance, Burden burden)
 			{
 				throw HelpfulExceptionsUtil.TrackInstanceCalledMultipleTimes(instance, burden);
 			}
+			finally
+			{
+				@lock.ExitWriteLock();
+			}
+
 			burden.Released += OnInstanceReleased;
 			perfCounter.IncrementTrackedInstancesCount();
 		}
 
 		private void OnInstanceReleased(Burden burden)
 		{
-			using (@lock.ForWriting())
+			try
 			{
+				@lock.EnterWriteLock();
```

Comment:
Same here, normally the lock entering code is placed before the `try`, but not here.

Reply:
Also, this (`OnInstanceReleased` taking a write lock) appears to support my suspicion that `Release` (above) should perhaps take an upgradeable read lock and do the `Release` call inside it.